traffic_ops/testing/api/v4/cachegroups_test.go

```diff
@@ -584,3 +586,67 @@ func CheckCacheGroupsAuthentication(t *testing.T) {
 		t.Error(fmt.Errorf(errFormat, "DeleteCacheGroupByID"))
 	}
 }
+
+func VerifyPaginationSupportCg(t *testing.T) {
+	qparams := url.Values{}
+	qparams.Set("orderby", "id")
+	cachegroup, _, err := TOSession.GetCacheGroupsByQueryParams(qparams)
+	if err != nil {
+		t.Fatalf("cannot GET Cachegroup: %v", err)
+	}
+
+	qparams = url.Values{}
+	qparams.Set("orderby", "id")
+	qparams.Set("limit", "1")
+	cachegroupWithLimit, _, err := TOSession.GetCacheGroupsByQueryParams(qparams)
+	if !reflect.DeepEqual(cachegroup[:1], cachegroupWithLimit) {
+		t.Error("expected GET Cachegroups with limit = 1 to return first result")
+	}
+
+	qparams = url.Values{}
+	qparams.Set("orderby", "id")
+	qparams.Set("limit", "1")
+	qparams.Set("offset", "1")
+	cachegroupsWithOffset, _, err := TOSession.GetCacheGroupsByQueryParams(qparams)
+	if !reflect.DeepEqual(cachegroup[1:2], cachegroupsWithOffset) {
+		t.Error("expected GET cachegroup with limit = 1, offset = 1 to return second result")
+	}
+
+	qparams = url.Values{}
+	qparams.Set("orderby", "id")
+	qparams.Set("limit", "1")
+	qparams.Set("page", "2")
+	cachegroupWithPage, _, err := TOSession.GetCacheGroupsByQueryParams(qparams)
+	if !reflect.DeepEqual(cachegroup[1:2], cachegroupWithPage) {
+		t.Error("expected GET cachegroup with limit = 1, page = 2 to return second result")
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "-2")
+	_, _, err = TOSession.GetCacheGroupsByQueryParams(qparams)
```

Comment:
Session.GetCacheGroupsByQueryParams() is deprecated, this should use Session.GetCacheGroupsByQueryParamsWithHdr()